types/record_def.go

```diff
@@ -96,7 +96,7 @@ func (r *RecordDefinition) publicSerializerMethodDef() string {
 }
 
 func (r *RecordDefinition) publicDeserializerMethodDef() string {
-	return fmt.Sprintf(recordStructPublicDeserializerTemplate, r.publicDeserializerMethod(), r.GoType(), r.deserializerMethod())
```

Comment:
Since we're replacing the concrete type with the interface name, I just put the interface in the actual template.